src/portal/src/app/project/project.module.ts

```diff
@@ -39,6 +39,10 @@ import { HelmChartModule } from './helm-chart/helm-chart.module';
 import { RobotAccountComponent } from './robot-account/robot-account.component';
 import { AddRobotComponent } from './robot-account/add-robot/add-robot.component';
 import { AddHttpAuthGroupComponent } from './member/add-http-auth-group/add-http-auth-group.component';
+import {TagRetentionComponent} from "./tag-retention/tag-retention.component";
+import {AddRuleComponent} from "./tag-retention/add-rule/add-rule.component";
+import {TagRetentionService} from "./tag-retention/tag-retention.service";
+import {EditRuleComponent} from "./tag-retention/edit-rule/edit-rule.component";
```

Comment:
WHY use different component `EditRuleComponent` and `AddRuleComponent` to cover the rule adding and editing?